import "VipGiftConfig.proto";
import "CommonGiftConfig.proto";

package ExploitBoxConfig;

//第三次及第五次获得宝箱数量为固定数量+随机额外数量
message ExploitRandomMul
{
	repeated uint32 stage_3 = 1;
	repeated uint32 stage_5 = 2;	
}

message ExploitReward
{
	optional VipGiftConfig.EquipReward	 equips 	= 1; //装备奖励
	optional CommonGiftConfig.BaseReward base  		= 2; //其他奖励  连同装备 只能配置一个奖励
	required bool						 is_mul 	= 3; //是否乘以概率倍数
	required uint32						 prob		= 4; //概率权值
}

message BaseRewardItem
{
	required CommonGiftConfig.BaseReward base = 1; //基本奖励
	required uint32						 prob = 2; //概率权值
}


message ExploitItem
{
	required uint32 		box_level 		= 1; //宝箱等级
	repeated uint32 		exploit_amount 	= 2; //各阶段的功勋
	repeated uint32 		box_cnt			= 3; //各阶段的开箱次数
	repeated ExploitReward	reward 			= 4;
	repeated BaseRewardItem base_reward		= 5;
}


message ExploitBox
{
	repeated uint32				prob 		=	1; //[1,2,4,10]的概率表
	required ExploitRandomMul	random_mul	=	2; //第三次及第五次随机额外数量 等概率
	repeated ExploitItem		out			= 	3;
}

message Exploit
{
	required ExploitBox exploit = 1;
}